### ISSUE TITLE
making shapes with incorrect points entry

### ISSUE THREAD

**Issue by Shimwell** [ukaea/paramak#694]
I noticed that the htc reactor example makes a RotateStraightShape with too many entries in the points arg and that this is not being caught by the points setter.

This should be caught by the code somewhere and should not be a valid input for the RotateStraightShape as it has splines in the connection type

```
import paramak

inner_vessel = paramak.RotateStraightShape(
    points=[
        (269.7459584295612, -46.54377880184336, 'straight'),
        (231.87066974595842, -46.5437788018433, 'spline'),
        (223.55658198614316, -62.21198156682027, 'spline'),
        (207.85219399538107, -80.64516129032262, 'spline'),
        (166.28175519630486, -115.66820276497697, 'spline'),
        (164.43418013856814, -119.35483870967744, 'spline'),
        (164.43418013856814, -122.11981566820276, 'straight'),
        (173.67205542725173, -140.5529953917051, 'straight'),
        (184.75750577367205, -140.5529953917051, 'straight'),
        (184.75750577367205, -158.98617511520735, 'straight'),
    ]
)


```

The result is that 'straight' is appended to each entry and the list ends up being 4 entries long

```

print(inner_vessel.points)

>> [(269.7459584295612, -46.54377880184336, 'straight', 'straight'), (231.87066974595842, -46.5437788018433, 'spline', 'straight'), (223.55658198614316, -62.21198156682027, 'spline', 'straight'), (207.85219399538107, -80.64516129032262, 'spline', 'straight'), (166.28175519630486, -115.66820276497697, 'spline', 'straight'), (164.43418013856814, -119.35483870967744, 'spline', 'straight'), (164.43418013856814, -122.11981566820276, 'straight', 'straight'), (173.67205542725173, -140.5529953917051, 'straight', 'straight'), (184.75750577367205, -140.5529953917051, 'straight', 'straight'), (184.75750577367205, -158.98617511520735, 'straight', 'straight'), (269.7459584295612, -46.54377880184336, 'straight', 'straight')]

```

**Comment by Shimwell** [comment]
This could be fixed by specifying the len of points in a additional (case specific) setter that runs in addition to the Shape points setter

```
    @RotateMixedShape.points.setter
    def points(self, values):
        if values is not None:
            # Checks that the length of each tuple in points is 2
            for value in values:
                if len(value) != 2:
                    msg = "individual points must contain 2 coordinates {} has a \
                        length of {}".format(value, len(values[0]))
                    raise ValueError(msg) 

        super(RotateStraightShape, self.__class__).points.fset(self, values)
```

**Comment by Shimwell** [comment]
Fixed with #695 

Perhaps the additional setters mentioned above could be used to simplify the points setter in Shape.py
